src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -21,6 +21,11 @@ namespace Microsoft.CodeAnalysis.CSharp.Extensions
 {
     internal static partial class SyntaxNodeExtensions
     {
+        public static void Deconstruct(this SyntaxNode node, out SyntaxKind kind)
+        {
+            kind = node.Kind();
+        }
```

Comment:
i dunno... seems too cute :)

Reply:
I had to check this PR out locally and load up VS to even understand what was going on :(

(not that VS helped.. apparently I missed the memo about deconstruct and patterns from C# 8 🤦‍♂️)

Reply:
Yeah this will probably goes into the "cute" bucket, per Cyrus terminology. But imagine any use-site with RawKind, I think this actually simplifies things quite a bit.